scipy/sparse/linalg/_isolve/_gcrotmk.py

```diff
@@ -283,11 +283,10 @@ def gcrotmk(A, b, x0=None, tol=1e-5, maxiter=1000, M=None, callback=None,
         raise ValueError("Invalid value for 'truncate': %r" % (truncate,))
 
     if atol is None:
-        warnings.warn("scipy.sparse.linalg.gcrotmk called without specifying `atol`. "
-                      "The default value will change in the future. To preserve "
-                      "current behavior, set ``atol=tol``.",
-                      category=DeprecationWarning, stacklevel=2)
         atol = tol
+    
+    if type(atol) != float:
+        raise ValueError("Tolerance value expected float")
```

Comment:
It should be possible to replace both if-conditions with
```suggestion
    if not np.issubdtype(type(atol), np.floating):
        raise ValueError("Parameter `atol` must be a floating-point number!")
```